src/main/java/io/phasetwo/service/model/jpa/InvitationAdapter.java

```diff
@@ -74,10 +74,16 @@ public void setUrl(String url) {
   @Override
   public UserModel getInviter() {
     if (invitation.getInviterId() != null) {
-      return session.users().getUserById(realm, invitation.getInviterId());
-    } else {
-      return null;
+      UserModel inviter = session.users().getUserById(realm, invitation.getInviterId());
+      // it is possible that the inviter was removed
+      if (inviter != null) {
+        return inviter;
+      } else {
+        // lazy remove the inviterId
+        invitation.setInviterId(null);
```

Comment:
We could add a e event to remove the invitation in the future. I saw the comment on the task